crates/ruma-events/src/error.rs

```diff
@@ -17,6 +17,7 @@ impl Error for InvalidInput {}
 
 /// An error when attempting to create a value from a string via the `FromStr` trait.
 #[derive(Clone, Eq, Debug, Hash, PartialEq)]
+#[allow(clippy::exhaustive_structs)]
```

Comment:
I think it would actually be better for this to be non-exhaustive. We might want to add more information to the error at a later point, and this type isn't meant to be created by other crates.